sklearn/feature_selection/univariate_selection.py

```diff
@@ -307,16 +306,415 @@ def f_regression(X, y, center=True):
 
 
 ######################################################################
-# Base classes
+# Base scorer class
+
+class _BaseScorer(BaseEstimator):
+    """Initialize the univariate feature scorer."""
+    def __init__(self):
+        pass
+
+    def fit(self, X, y):
```

Comment:
This seems inappropriate

Reply:
How much do I need in the base class?

Reply:
You can have `pass` in the base class. Or have no base class. Or have something like this, but annotate with abstractmethod

Reply:
The reason I have a _BaseScorer class is to be able able to check in _BaseFilter that a passed instance in score_func is correct.